src/App.js

```diff
@@ -1,24 +1,54 @@
-import logo from './logo.svg';
-import './App.css';
+import StyledHeader from "./styled/StyledHeader";
+import StyledList from "./styled/StyledList";
+import StyledButton from "./styled/StyledButton";
+import StyledSearchBar from "./styled/StyledSearchBar";
+import { useEffect, useState } from "react";
 
 function App() {
+  const apiUrl = "https://fetch-me.vercel.app/api/shopping/items";
+  const [products, setProducts] = useState([]);
+  const [productSearch, setProductSearch] = useState("");
+
+  useEffect(() => {
+    function loadData() {
+      fetch(apiUrl)
+        .then((response) => response.json())
+        .then(({ data }) => {
+          setProducts(data);
+        });
+    }
+
+    loadData();
+  }, [apiUrl]);
+
+  function onSubmit(event) {
+    event.preventDefault();
+    setProductSearch(event.target.value);
+  }
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
+      <StyledButton>DE</StyledButton>
+      <StyledButton>EN</StyledButton>
+      <StyledHeader>Shopping List</StyledHeader>
+      <StyledSearchBar>
+        <label hmtlFor="header-search">
+          <span>Was möchtest du kaufen?</span>
+        </label>
+        <input
+          type="text"
+          id="header-search"
+          name="search"
+          onChange={onSubmit}
+        />
+      </StyledSearchBar>
+      <StyledList>
+        {products
+          .filter((product) => product.name.de.includes(productSearch))
+          .map((product) => (
+            <li key={product._id}>{product.name.de}</li>
+          ))}
+      </StyledList>
+    </>
```

Comment:
Nice job :) instead of just having the components as exported `styled-components` (`StyledList`, `StyledSearchBar`), I would recommend to extract the structures here and add them to the corresponding component. For example, the `SearchBar` could be its own component as well as the `List`. Doing this would clean up the App.js a bit :)